lib/storage/src/db/replay.rs

```diff
@@ -288,24 +234,34 @@ impl ReadReplay for BlockReplayStorage {
             block_output_hash: B256::from_slice(&block_output_hash),
         })
     }
+
+    fn latest_record(&self) -> BlockNumber {
+        // This is guaranteed to be non-`None` because genesis is always inserted on storage initialization.
+        self.latest_record_checked()
+            .expect("no blocks in BlockReplayStorage")
+    }
 }
 
 impl WriteReplay for BlockReplayStorage {
-    fn append(&self, record: ReplayRecord) {
+    fn append(&self, record: ReplayRecord) -> bool {
         let latency_observer = BLOCK_REPLAY_ROCKS_DB_METRICS.get_latency.start();
-        assert!(!record.transactions.is_empty());
-
-        let current_latest_block = self.latest_block().unwrap_or(0);
-
-        if record.block_context.block_number <= current_latest_block {
+        let current_latest_record = self.latest_record();
+        if record.block_context.block_number <= current_latest_record {
             tracing::debug!(
-                "Not appending block {}: already exists in WAL",
-                record.block_context.block_number
+                block_number = record.block_context.block_number,
+                "not appending block: already exists in WAL",
```

Comment:
nit: when I wrote this I wasn't consistent with namings.. Sometimes `WAL` sometimes `Block Replay Storage`. I'd use the latter everywhere. Can be tackled in a separete PR (I can do it)

Reply:
I renamed everything apart from column family